### PR TITLE
docs: translate basics>extend.md

### PR DESCRIPTION
**Pull request by DanielWLam**
##### Checklist

- [x] documentation is changed or added

##### Affected core subsystem(s)

`docs>source>en>service.md`

### CODE REVIEW

File: docs/source/en/basics/extend.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    // this.ctx => context object
+    // this.app => application object
+  },
+};
+```
+
+
+## Extend according to environment
+
+Besides, we can extend the framework in an optional way according to the environment. For example, if you want `mockXX()` only be able to accessed when doing unittest:
```
Comment (dead-horse) [inline docs/source/en/basics/extend.md]: when doing unittest => in unittest environment

File: docs/source/en/basics/extend.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+// app/extend/application.unittest.js
+module.exports = {
+  mockXX(k, v) {
+  }
+};
+```
+
+This file will only be required under unittest environment.
+
+Similarly, we could extend egg in this way for orther object,such as Application, Context, Request, Response and Helper. See more on [environment](./env.md)
```
Comment (dead-horse) [inline docs/source/en/basics/extend.md]: other objects

**Comment by dead-horse** [comment]
没什么大问题的话可以先合，感觉中文部分也有一些问题，到时候再统一优化一下。

**Comment by fengmk2** [comment]
Commit log must be English.

**Comment by popomore** [comment]
> Commit log must be English.

We can ignore these commit message when merge.

**Comment by popomore** [comment]
@DanielWLam can you see the comment from @dead-horse ?

**Comment by popomore** [comment]
https://github.com/eggjs/egg/issues/363

**Comment by popomore** [comment]
@dead-horse 我先合了，这些意见可以再提 MR 改

**Comment by DanielWLam** [comment]
@popomore how could i change the commit log which have already been commit?

**Comment by atian25** [comment]
just rebase, but since we will merge PR with squash merge, so if you don't familiar with rebase, you could just left it.

发自我的 iPhone

**Pull request merged**